### PR TITLE
Do OPM form lookups in bulk

### PR DESCRIPTION
**Pull request by esoergel**
@czue @sravfeyn
Low-hanging fruit that could bump performance.  Gains don't seem to be much locally, but prod feels db overhead a lot more than I do.

### CODE REVIEW

File: custom/opm/reports.py [hunk 1]
```diff
@@ -309,6 +314,19 @@ def get_dates_in_range(self, owner_id, startdate, enddate, prop='vhnd_available'
...
+    def forms_by_case(self):
+        assert self.cases is not None, \
+            "SharedDataProvider was not instantiated with cases"
+        all_form_ids = chain(*(case.xform_ids for case in self.cases))
+        forms_by_case = defaultdict(list)
+        for form in iter_docs(XFormInstance.get_db(), all_form_ids):
+            if form['xmlns'] in OPM_XMLNSs:
+                case_id = form['form']['case']['@case_id']
+                forms_by_case[case_id].append(XFormInstance.wrap(form))
+        return forms_by_case
```
Comment (esoergel) [inline custom/opm/reports.py]: This is the meat of everything.

**Comment by czue** [comment]
:+1: when tests pass

**Comment by czue** [comment]
@nickpell I thought you fixed this:
```
======================================================================
FAIL: test_midnight_for_domain (corehq.apps.callcenter.tests.test_utils.DomainTimezoneTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/callcenter/tests/test_utils.py", line 164, in test_midnight_for_domain
    self.assertEqual(dom.midnights[0], midnight_past - timedelta(hours=offset), tz)
AssertionError: datetime.datetime(2015, 8, 25, 18, 30) != datetime.datetime(2015, 8, 26, 18, 30) : Asia/Kolkata
```

**Comment by czue** [comment]
(i just kicked build before it failed again so definitely started after https://github.com/dimagi/commcare-hq/pull/8009 was merged)

**Comment by nickpell** [comment]
reproduced and looking

File: custom/opm/reports.py [hunk 2]
```diff
@@ -309,6 +314,19 @@ def get_dates_in_range(self, owner_id, startdate, enddate, prop='vhnd_available'
             [date for date in self._service_dates[owner_id][prop]],
         )
 
+    @property
+    @memoized
+    def forms_by_case(self):
+        assert self.cases is not None, \
+            "SharedDataProvider was not instantiated with cases"
+        all_form_ids = chain(*(case.xform_ids for case in self.cases))
```
Comment (sravfeyn) [inline custom/opm/reports.py]: sweet!

**Pull request merged**